base/strings/util.jl

```diff
@@ -502,10 +502,8 @@ See also [`split`](@ref).
 julia> a = "Ma.rch"
 "Ma.rch"
 
-julia> collect(eachsplit(a, "."))
-2-element Vector{SubString}:
- "Ma"
- "rch"
+julia> NTuple{2}(eachsplit(a, '.'))
+("Ma", "rch")
```

Comment:
Maybe `foreach` is nicer than `collect` as a way to consume the whole iterator, something like:
```
julia> a = "Ma.r.ch";

julia> foreach(println, eachsplit(a, '.'))
Ma
r
ch

julia> NTuple{2}(eachsplit(a, '.'))
("Ma", "r")
```